include/swift/Sema/CSFix.h

```diff
@@ -2648,6 +2659,49 @@ class TreatEphemeralAsNonEphemeral final : public AllowArgumentMismatch {
   }
 };
 
+/// Error if a user passes let f: (sending T) -> () as a (T) -> ().
+///
+/// This prevents data races since f assumes its parameter if the parameter is
+/// non-Sendable is safe to transfer onto other situations. The caller though
+/// that this is being sent to does not enforce that invariants within its body.
+class AllowSendingMismatch final : public ContextualMismatch {
+public:
+  enum class Kind {
```

Comment:
I don't think this is necessary, it could be determined based on the locator associated wit the fix.